kqueen/blueprints/api/views.py

```diff
@@ -31,86 +55,313 @@ def index():
 
 
 # Clusters
+
 @api.route('/clusters', methods=['GET'])
 @jwt_required()
 def cluster_list():
-    # TODO: implement native serialization
-
     output = []
 
-    for cluster in list(Cluster.list(return_objects=True).values()):
-        output.append(cluster.get_dict())
+    for obj in list(Cluster.list(return_objects=True).values()):
+        output.append(obj.get_dict(expand=True))
+
+    return jsonify(output)
+
+
+@api.route('/clusters', methods=['POST'])
+@jwt_required()
+def cluster_create():
+    if not request.json:
+        abort(400)
+    else:
+        obj = Cluster(**request.json)
+        try:
+            # save cluster
+            obj.save()
+            output = obj.get_dict(expand=True)
+
+            # start provisioning
+            obj.engine.provision()
+
+        except Exception as e:
+            logger.error(e)
+            abort(500)
 
     return jsonify(output)
 
 
-@api.route('/clusters/<cluster_id>', methods=['GET'])
+@api.route('/clusters/<pk>', methods=['GET'])
 @jwt_required()
-def cluster_detail(cluster_id):
+def cluster_get(pk):
+    obj = get_object(Cluster, pk)
+
+    return jsonify(obj.get_dict(expand=True))
+
+
+@api.route('/clusters/<pk>', methods=['PATCH'])
+@jwt_required()
+def cluster_update(pk):
+    if not request.json:
+        abort(400)
+
+    data = request.json
```

Comment:
It shouldn't be necessary to test if request.json if dict after you tested it exists

Reply:
We need it to be dict of names and it's valued.
For example `[1, 2, 3]` is valid JSON but it doesn't works.